src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1677,31 +1720,38 @@ protected <T> void deleteTaskDocs(
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         ADTask adTask = ADTask.parse(parser, searchHit.getId());
                         logger.debug("Delete old task: {} of detector: {}", adTask.getTaskId(), adTask.getDetectorId());
-                        bulkRequest.add(new DeleteRequest(DETECTION_STATE_INDEX).id(adTask.getTaskId()));
+                        operations
+                            .add(new BulkOperation.Builder().delete(d -> d.index(DETECTION_STATE_INDEX).id(adTask.getTaskId())).build());
                     } catch (Exception e) {
                         listener.onFailure(e);
                     }
                 }
-                client.execute(BulkAction.INSTANCE, bulkRequest, ActionListener.wrap(res -> {
+
+                BulkRequest bulkRequest = new BulkRequest.Builder().operations(operations).build();
+                try {
+                    CompletableFuture<BulkResponse> bulkResponse = sdkJavaAsyncClient.bulk(bulkRequest);
+                    BulkResponse res = bulkResponse.orTimeout(10L, TimeUnit.SECONDS).get();
+
                     logger.info("Old AD tasks deleted for detector {}", detectorId);
-                    BulkItemResponse[] bulkItemResponses = res.getItems();
-                    if (bulkItemResponses != null && bulkItemResponses.length > 0) {
-                        for (BulkItemResponse bulkItemResponse : bulkItemResponses) {
-                            if (!bulkItemResponse.isFailed()) {
-                                logger.debug("Add detector task into cache. Task id: {}", bulkItemResponse.getId());
+                    List<BulkResponseItem> bulkItemResponses = res.items();
+                    if (bulkItemResponses != null && bulkItemResponses.size() > 0) {
+                        for (BulkResponseItem bulkItemResponse : bulkItemResponses) {
+                            if (bulkItemResponse.error() == null) {
```

Comment:
Nit: Verify once. Isn't the condition the other way round?

Reply:
Correct, let me change this